src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2358,6 +2365,35 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         return _core->TaskbarProgress();
     }
 
+    void TermControl::BellLightOn()
+    {
+        if (!_bellLightTimer)
+        {
+            // Start the timer, when the timer ticks we switch off the light
+            DispatcherTimer invertTimer;
+            invertTimer.Interval(std::chrono::milliseconds(TerminalWarningBellInterval));
+            invertTimer.Tick({ get_weak(), &TermControl::_BellLightOff });
+            invertTimer.Start();
+            _bellLightTimer.emplace(std::move(invertTimer));
+
+            // Switch on the light and animate the intensity to fade out
+            VisualBellLight::SetIsTarget(RootGrid(), true);
+            BellLight().CompositionLight().StartAnimation(L"Intensity", _bellLightAnimation);
+        }
+    }
+
+    void TermControl::_BellLightOff(Windows::Foundation::IInspectable const& /* sender */,
+                                    Windows::Foundation::IInspectable const& /* e */)
+    {
+        if (_bellLightTimer && !_closing)

```

Comment:
I believe you should theoretically also stop the timer if the application is closing. You can also use the `->` operator instead of `.value()` and `.reset()` for clearing the optional.
For instance:

```cpp
if (!_bellLightTimer)
{
	return;
}

_bellLightTimer->Stop();
_bellLightTimer.reset();

if (_closing)
{
	return;
}

VisualBellLight::SetIsTarget(RootGrid(), false);
```

(I personally like "linear" flows like this, but it would be just as fine if you write it with nested if conditions.)

Reply:
I personally don't like having `return`s in the middle, changed it so we stop the timer even if closing though